aws/resource_aws_gamelift_fleet_test.go

```diff
@@ -0,0 +1,625 @@
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "operating_system", "WINDOWS_2012"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "resource_creation_limit_policy.#", "1"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "resource_creation_limit_policy.0.new_game_sessions_per_creator", "4"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "resource_creation_limit_policy.0.policy_period_in_minutes", "25"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.#", "1"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.0.game_session_activation_timeout_seconds", "35"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.0.max_concurrent_game_session_activations", "98"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.0.server_process.#", "1"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.0.server_process.0.concurrent_executions", "1"),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.0.server_process.0.launch_path", launchPath),
+					resource.TestCheckResourceAttr("aws_gamelift_fleet.test", "runtime_configuration.0.server_process.0.parameters", params[1]),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSGameliftFleetExists(n string, res *gamelift.FleetAttributes) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No Gamelift Fleet ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).gameliftconn
+
+		out, err := conn.DescribeFleetAttributes(&gamelift.DescribeFleetAttributesInput{
+			FleetIds: aws.StringSlice([]string{rs.Primary.ID}),
+		})
+		if err != nil {
+			return err
+		}
+		attributes := out.FleetAttributes
+		if len(attributes) < 1 {
+			return fmt.Errorf("Gamelift Fleet %q not found", rs.Primary.ID)
+		}
+		if len(attributes) != 1 {
+			return fmt.Errorf("Expected exactly 1 Gamelift Fleet, found %d under %q",
+				len(attributes), rs.Primary.ID)
+		}
+		fleet := attributes[0]
+
+		if *fleet.FleetId != rs.Primary.ID {
+			return fmt.Errorf("Gamelift Fleet not found")
+		}
+
+		*res = *fleet
+
+		return nil
+	}
+}
+
+func testAccCheckAWSGameliftFleetDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).gameliftconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_gamelift_fleet" {
+			continue
+		}
+
+		out, err := conn.DescribeFleetAttributes(&gamelift.DescribeFleetAttributesInput{
+			FleetIds: aws.StringSlice([]string{rs.Primary.ID}),
+		})
+		if err != nil {
+			return err
+		}
+
+		attributes := out.FleetAttributes
+
+		if len(attributes) > 0 {
```

Comment:
Do we need to check its status for `gamelift.FleetStatusDeleting` or `gamelift.FleetStatusTerminated`?

Reply:
`waitForGameliftFleetToBeDeleted` should guarantee that we don't as the fleet should be completely gone by this time. If it's not I'd rather treat that as bug.

Reply:
Ah sorry, I missed this when looking at `waitForGameliftFleetToBeDeleted` - we're going for zero returned results instead of just `gamelift.FleetStatusTerminated` status. I thought you'd be getting the one result with the terminated status 👍